### PR TITLE
FEAT(client): Adds a new shortcut which hides/shows the main window

### PR DESCRIPTION
**Pull request by merua**
Implements #2328

### CODE REVIEW

File: src/mumble/MainWindow.cpp [hunk 1]
```diff
@@ -3027,6 +3030,22 @@ void MainWindow::on_gsCycleTransmitMode_triggered(bool down, QVariant) {
 	updateTransmitModeComboBox();
 }
 
+void MainWindow::on_gsToggleMainWindowVisibility_triggered(bool down, QVariant) {
+	if(down)
+	{
+		if(g.mw->isVisible())
+		{
```
Comment (davidebeatrici) [inline src/mumble/MainWindow.cpp:3037]:
```suggestion
	if (down) {
		if (g.mw->isVisible()) {
```

File: src/mumble/MainWindow.cpp [hunk 2]
```diff
@@ -3027,6 +3030,22 @@ void MainWindow::on_gsCycleTransmitMode_triggered(bool down, QVariant) {
...
+			g.mw->hide();
+			g.l->log(Log::DebugInfo, tr("Main window hidden."));
+		}
+		else
+		{
```
Comment (davidebeatrici) [inline src/mumble/MainWindow.cpp:3042]:
```suggestion
		} else {
```

**Review by davidebeatrici — CHANGES_REQUESTED** [review]
I would also remove the log messages.

**Review by Krzmbrzl — COMMENTED** [review]
Furthermore you have to update the translations to also include your new string.

In order to do so, please run `scripts/updatetranslations.sh` :point_up:

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]
Thanks for the changes!

Now I'd only ask you to squash your first 2 commits (but keep the translation one separate) into a single one (the commit history of how the review has influenced the code is nothing that has to go into master).

After that I think we're all set :+1:

File: src/mumble/MainWindow.cpp (hunk 1 above)
Comment (davidebeatrici) [reply src/mumble/MainWindow.cpp]: Still the case.

**Review by Krzmbrzl — APPROVED** [review]

File: src/mumble/MainWindow.cpp [hunk 3]
```diff
@@ -3027,6 +3030,16 @@ void MainWindow::on_gsCycleTransmitMode_triggered(bool down, QVariant) {
 	updateTransmitModeComboBox();
 }
 
+void MainWindow::on_gsToggleMainWindowVisibility_triggered(bool down, QVariant) {
+	if(down) {
+		if(g.mw->isVisible()) {
```
Comment (davidebeatrici) [inline src/mumble/MainWindow.cpp:3035]:
```suggestion
	if (down) {
		if (g.mw->isVisible()) {
```

**Review by davidebeatrici — APPROVED** [review]

**Comment by davidebeatrici** [comment]
Thank you very much for your contribution!

**Comment by merua** [comment]
Thanks to you and all the other contributors.

I have started using Mumble since the lockdowns and it has really helped staying in touch with my friends, so keep up the good work! (Especially since I found out -- to my horror -- that a lot of problem tickets are written in a very toxic tone.)

And next time I will try to set up the formatter. Sorry for that...

**Comment by davidebeatrici** [comment]
Thanks!

No worries about the formatting, we should probably add a CI task that runs `clang-format` and then shows the changes.

**Pull request merged**